### PR TITLE
enhance: vlite as video/audio player

### PR DESCRIPTION
**Pull request by ThatOneCalculator**
# What
Uses VLiteJS as player

# Why
Current player is very basic

# TODO

- [ ] Fix overlapping
- [ ] PIP

### CODE REVIEW

**Comment by ThatOneCalculator** [comment]
Currently drafted because of https://github.com/vlitejs/vlite/issues/68

**Comment by ThatOneCalculator** [comment]
Closer... now they overlap!

image

File: packages/client/package.json [hunk 1]
```diff
@@ -108,5 +109,6 @@
 		"eslint-plugin-import": "2.26.0",
 		"eslint-plugin-vue": "9.2.0",
 		"start-server-and-test": "1.14.0"
-	}
+	},
+	"packageManager": "yarn@3.2.1"
```
Comment (acid-chicken) [inline packages/client/package.json]:
```suggestion
	}
```

File: packages/client/src/components/media-video.vue [hunk 2]
```diff
@@ -7,38 +7,49 @@
 </div>
 <div v-else class="kkjnbbplepmiyuadieoenjgutgcmtsvu">
 	<video
+		id="player"
```
Comment (acid-chicken) [inline packages/client/src/components/media-video.vue]:
```suggestion
```

File: packages/client/src/components/media-video.vue [hunk 3]
```diff
@@ -7,38 +7,49 @@
...
+import 'vlitejs/dist/vlite.css';
+import Vlitejs from 'vlitejs';
 import { defaultStore } from '@/store';
 
 const props = defineProps<{
 	video: misskey.entities.DriveFile;
 }>();
 
+onMounted(() => {
+	new Vlitejs('#player');
```
Comment (acid-chicken) [inline packages/client/src/components/media-video.vue]:
```suggestion
const videoEl = $ref(null);

onMounted(() => {
	new Vlitejs(videoEl);
```

File: packages/client/src/components/media-banner.vue [hunk 4]
```diff
@@ -6,8 +6,10 @@
 		<span>{{ $ts.clickToShow }}</span>
 	</div>
 	<div v-else-if="media.type.startsWith('audio') && media.type !== 'audio/midi'" class="audio">
-		<audio ref="audioEl"
-			class="audio"
+		<audio
+			id="player"
```
Comment (acid-chicken) [inline packages/client/src/components/media-banner.vue]:
```suggestion
```

File: packages/client/src/components/media-banner.vue [hunk 5]
```diff
@@ -43,7 +47,10 @@ function volumechange() {
 }
 
 onMounted(() => {
-	if (audioEl) audioEl.volume = ColdDeviceStorage.get('mediaVolume');
+	if (audioEl) {
+		audioEl.volume = ColdDeviceStorage.get('mediaVolume');
+		new Vlitejs('#player');
```
Comment (acid-chicken) [inline packages/client/src/components/media-banner.vue]:
```suggestion
		new Vlitejs(audioEl);
```

File: packages/client/package.json (hunk 1 above)
Comment (ThatOneCalculator) [reply packages/client/package.json]: Sorry about that, yarn did it and i didn't catch that

**Comment by ThatOneCalculator** [comment]
Closing, I now have plyr working in Calckey. Might PR here.

**Pull request closed without merging**